Lib9c/Model/Quest/QuestList.cs

```diff
@@ -361,58 +389,62 @@ public CollectionMap UpdateCompletedQuest(CollectionMap eventMap)
-            out Quest quest)
+            out (Quest quest, string errorMessage) result)
         {
+            result = (null, string.Empty);
             switch (row)
             {
+                default:
+                    result.quest = null;
+                    result.errorMessage = $"Unexpected type: {row.GetType().FullName}";
+                    break;
                 case CollectQuestSheet.Row r:
-                    quest = new CollectQuest(r, reward);
+                    result.quest = new CollectQuest(r, reward);
                     break;
                 case CombinationQuestSheet.Row r:
-                    quest = new CombinationQuest(r, reward);
+                    result.quest = new CombinationQuest(r, reward);
                     break;
                 case GeneralQuestSheet.Row r:
-                    quest = new GeneralQuest(r, reward);
+                    result.quest = new GeneralQuest(r, reward);
                     break;
                 case ItemEnhancementQuestSheet.Row r:
-                    quest = new ItemEnhancementQuest(r, reward);
+                    result.quest = new ItemEnhancementQuest(r, reward);
                     break;
                 case ItemGradeQuestSheet.Row r:
-                    quest = new ItemGradeQuest(r, reward);
+                    result.quest = new ItemGradeQuest(r, reward);
                     break;
                 case MonsterQuestSheet.Row r:
-                    quest = new MonsterQuest(r, reward);
+                    result.quest = new MonsterQuest(r, reward);
                     break;
                 case TradeQuestSheet.Row r:
-                    quest = new TradeQuest(r, reward);
+                    result.quest = new TradeQuest(r, reward);
                     break;
                 case WorldQuestSheet.Row r:
-                    quest = new WorldQuest(r, reward);
+                    result.quest = new WorldQuest(r, reward);
                     break;
                 case ItemTypeCollectQuestSheet.Row r:
-                    quest = new ItemTypeCollectQuest(r, reward);
+                    result.quest = new ItemTypeCollectQuest(r, reward);
                     break;
                 case GoldQuestSheet.Row r:
-                    quest = new GoldQuest(r, reward);
+                    result.quest = new GoldQuest(r, reward);
                     break;
                 case CombinationEquipmentQuestSheet.Row r:
                     int stageId;
                     var recipeRow = equipmentItemRecipeSheet.Values
                         .FirstOrDefault(e => e.Id == r.RecipeId);
                     if (recipeRow is null)
                     {
-                        throw new ArgumentException($"Invalid Recipe Id : {r.RecipeId}");
+                        result.quest = null;
+                        result.errorMessage = $"Invalid Recipe Id : {r.RecipeId}";
+                        break;
```

Comment:
i think this change can be different execution result in old blocks.

Reply:
Yes.. I missed that point.🙃